app/src/main/java/de/xikolo/controllers/helper/ShortcutHelper.kt

```diff
@@ -0,0 +1,50 @@
+package de.xikolo.controllers.helper
+
+import android.content.Context
+import android.content.Intent
+import android.content.pm.ShortcutInfo
+import android.content.pm.ShortcutManager
+import android.graphics.drawable.Icon
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat.getSystemService
+import de.xikolo.R
+import de.xikolo.controllers.course.CourseActivityAutoBundle
+import de.xikolo.storages.RecentCoursesStorage
+
+class ShortcutHelper {
+
+    @RequiresApi(Build.VERSION_CODES.N_MR1)
+    fun addCourse(context: Context, courseId: String, title: String) {
+
+        val recentCoursesStorage = RecentCoursesStorage()
+        recentCoursesStorage.addCourse(courseId, title)
+
+        updateShortcuts(context)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.N_MR1)
```

Comment:
Remove the `RequiresApi` annotations. Move all the code that requires a certain API version to an own method and use `TargetApi` there. The other helper methods which call these method(s) do the feature check before. Other parts of the app don't care so let's do that here.